code/script.js

```diff
@@ -1,19 +1,147 @@
+    alert(`Please enter a valid number: 1, 2, or 3`)
+  }
+}
+
+//The first order confirmation
+
+if (firstOrder === '1') {
+  firstOrder = 'Pizza'
+}
+else if (firstOrder === '2') {
+  firstOrder = 'Pasta'
+}
+else if (firstOrder === '3') {
+  firstOrder = 'Salad'
+}
+
+alert(
+  `You would like to have ${firstOrder}. Is that correct?`
+)
+
+// The second order
+
+//Pizza Type
+secondOrder = ''
+if (firstOrder === 'Pizza') {
+  while (secondOrder !== '1' && secondOrder !== '2' && secondOrder !== '3') {
+    secondOrder = prompt(`What type of Pizza would you like to order? Please write the number:
+  1 - Pizza Margherita
+  2 - Pizza Pepperoni
+  3 - Pizza Hawaiian`)
+    if (secondOrder !== '1' && secondOrder !== '2' && secondOrder !== '3') {
+      alert(`Pease enter a valid number: 1, 2, or 3`)
+    }
+  }
+
+
+  if (secondOrder === '1') {
+    secondOrder = 'Pizza Margherita'
+  }
 
-// Step 3 - Subtype choice
-// Your code goes here
+  else if (secondOrder === '2') {
+    secondOrder = 'Pizza Pepperoni'
+  }
+
+  else if (secondOrder === '3') {
+    secondOrder = 'Pizza Hawaiian'
+  }
+}
+
+
+// Pasta Type
+
+else if (firstOrder === 'Pasta') {
+  while (secondOrder !== '1' && secondOrder !== '2' && secondOrder !== '3') {
+    secondOrder = prompt(`What kind of Pasta would you like to order today? Please write the number:
+    1 - Spaghetti
+    2 - Carbonara
+    3 - Lasagna`)
+    if (secondOrder !== '1' && secondOrder !== '2' && secondOrder !== '3') {
+      alert(`Please enter a valid number: 1, 2, or 3!`)
+    }
+  }
+
+  if (secondOrder === '1') {
+    secondOrder = 'Spaghetti'
+  }
+  else if (secondOrder === '2') {
+    secondOrder = 'Carbonara'
+  }
+  else if (secondOrder === '3') {
+    secondOrder = 'Lasagna'
+  }
+}
+
+// Salad Type
+
+else if (firstOrder === 'Salad') {
+  while (secondOrder !== '1' && secondOrder !== '2' && secondOrder !== '3') {
+    secondOrder = prompt(`What kind of Salad would you love to order today? Please write the number:
+    1 - Greek Salad
+    2 - Fish Salad
+    3 - Chicken Salad`)
+    if (secondOrder !== '1' && secondOrder !== '2' && secondOrder !== '3') {
+      alert(`Please enter a valid number: 1, 2, or 3!`)
+    }
+  }
+
+
+  if (secondOrder === '1') {
+    secondOrder = 'Greek Salad'
+  }
+  else if (secondOrder === '2') {
+    secondOrder = 'Fish Salad'
+  }
+  else if (secondOrder === '3') {
+    secondOrder = 'Chicken Salad'
+  }
+}
+
+alert(
+  `${secondOrder}, great choice!`
+)
 
 // Step 4 - Age
-// Your code goes here
+let age = prompt(`To know the right size of your meal one last question: How old are you, ${name}?`)
+
+// Check if age is not a number
+while (isNaN(age)) {
+  alert(`Please enter a valid number`)
+  age = prompt(`Enter your age:`)
+}
+//Order 
+if (age >= 13) {
+  alert(`Thank you! You like to order ${secondOrder} regular size. Click 'Ok' to order.`)
+}
+else if (age < 13) {
+  alert(`Thank you.You like to order a kid's ${secondOrder}. Click 'ok' to order`)
+}
+
```

Comment:
The only thing I found which maybe could have an improvement at this stage is that you could parse the age value and validate against potential edge cases (e.g., negative or unrealistic ages).

let age = parseInt(prompt(`How old are you, ${name}?`));
while (isNaN(age) || age < 0 || age > 120) {
    alert(`Please enter a valid age.`);
    age = parseInt(prompt(`Enter your age:`));
}